lib/cocina/models.rb

```diff
@@ -49,6 +49,31 @@ class ValidationError < Error; end
     class Struct < Dry::Struct
       transform_keys(&:to_sym)
       schema schema.strict
+
+      def self.new(attributes = default_attributes, safe = false, &block)
+        super(underscore_attributes(attributes), safe, &block)
+      end
```

Comment:
Adding this constructor to the `Struct` superclass, from which all Cocina models inherit, allows for conversion of camelCase keys to snake_case ones at instantiation time.